src/legacy/core_plugins/data/public/search/search_bar/components/search_bar.tsx

```diff
@@ -133,10 +163,10 @@ class SearchBarUI extends Component<Props, State> {
         onClick={this.toggleFiltersVisible}
         isSelected={this.state.isFiltersVisible}
         hasActiveFilters={this.state.isFiltersVisible}
-        numFilters={this.props.filters.length > 0 ? this.props.filters.length : undefined}
+        numFilters={this.getFilterLength() ? this.getFilterLength() : undefined}
         aria-controls="GlobalFilterGroup"
         aria-expanded={!!this.state.isFiltersVisible}
-        title={`${this.props.filters.length} ${filtersAppliedText} ${clickToShowOrHideText}`}
+        title={`${this.getFilterLength()} ${filtersAppliedText} ${clickToShowOrHideText}`}
```

Comment:
What happens to this string if `getFilterLength` returns `void`? (Is that even possible?)

Reply:
Fixed

Reply:
I guess what I meant here was that `getFilterLength` looks like this:
```ts
  private getFilterLength() {
    if (this.props.showFilterBar && this.props.filters) {
      return this.props.filters.length;
    }
  }
```

so if the conditional evaluates to false, the function won't return (`void`). what happens then in your template string?

```js
`${this.getFilterLength()} ${filtersAppliedText} ${clickToShowOrHideText}`
```

is `getFilterLength` just going to be an empty string? this is mostly a question, i honestly don't know and haven't tested

Reply:
The message is used like this:

![image](https://user-images.githubusercontent.com/3016806/61443128-92b08d80-a951-11e9-80ab-2443764fefc2.png)

So you're right. I fixed to show only "select to show\hide" when filter count is 0.